### PR TITLE
sm7125: Ascendia, ascend!

### PR DESCRIPTION
**Pull request by RisenID**
Self promotion bess lol

### CODE REVIEW

File: target/a52q/patches/ascendia/customize.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    if [ -e "$FILE_PATH" ]; then
+        local FILE
+        local PARTITION
+        FILE="$(echo -n "$FILE_PATH" | sed "s.$WORK_DIR/..")"
+        PARTITION="$(echo -n "$FILE" | cut -d "/" -f 1)"
+
+        echo "Debloating /$FILE"
+        rm -rf "$FILE_PATH"
+    fi
+}
```
Comment (salvogiangri) [inline target/a52q/patches/ascendia/customize.sh]: This can be removed

File: target/a72q/patches/ascendia/customize.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    if [ -e "$FILE_PATH" ]; then
+        local FILE
+        local PARTITION
+        FILE="$(echo -n "$FILE_PATH" | sed "s.$WORK_DIR/..")"
+        PARTITION="$(echo -n "$FILE" | cut -d "/" -f 1)"
+
+        echo "Debloating /$FILE"
+        rm -rf "$FILE_PATH"
+    fi
+}
```
Comment (salvogiangri) [inline target/a72q/patches/ascendia/customize.sh]: This can be removed

File: target/a72q/patches/ascendia/module.prop [hunk 3]
```diff
@@ -0,0 +1,4 @@
+id=ascendia
+name=Ascendia Kernel support
+author=BlackMesa123 & RisenID
```
Comment (salvogiangri) [inline target/a72q/patches/ascendia/module.prop]: Only you in the credits would be better imo

File: target/a52q/patches/ascendia/module.prop [hunk 4]
```diff
@@ -0,0 +1,4 @@
+id=ascendia
+name=Ascendia Kernel support
+author=BlackMesa123 & RisenID
```
Comment (salvogiangri) [inline target/a52q/patches/ascendia/module.prop]: Only you in the credits would be better imo

File: target/a72q/patches/ascendia/customize.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+{
+    [ -d "$TMP_DIR" ] && rm -rf "$TMP_DIR"
+    mkdir -p "$TMP_DIR"
+
+    echo "Downloading $(basename "$ASCENDIA_IMG")"
+    curl -L -s -o "$TMP_DIR/boot.img" "$ASCENDIA_IMG"
+
+    echo "Extracting kernel binaries"
+    rm -f "$WORK_DIR/kernel/"*.img
+    cp "$TMP_DIR/boot.img" "$WORK_DIR/kernel/"
```
Comment (salvogiangri) [inline target/a72q/patches/ascendia/customize.sh]: Remove only boot.img from the kernel dir/adjust cp cmd:
https://github.com/BlackMesa123/UN1CA/blob/967b129dfa4823e380da02033d00c553ef4cb325/target/a54x/patches/kernelsu/customize.sh#L15-L18

File: target/a52q/patches/ascendia/customize.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+{
+    [ -d "$TMP_DIR" ] && rm -rf "$TMP_DIR"
+    mkdir -p "$TMP_DIR"
+
+    echo "Downloading $(basename "$ASCENDIA_IMG")"
+    curl -L -s -o "$TMP_DIR/boot.img" "$ASCENDIA_IMG"
+
+    echo "Extracting kernel binaries"
+    rm -f "$WORK_DIR/kernel/"*.img
+    cp "$TMP_DIR/boot.img" "$WORK_DIR/kernel/"
```
Comment (salvogiangri) [inline target/a52q/patches/ascendia/customize.sh]: Remove only boot.img from the kernel dir/adjust cp cmd:
https://github.com/BlackMesa123/UN1CA/blob/967b129dfa4823e380da02033d00c553ef4cb325/target/a54x/patches/kernelsu/customize.sh#L15-L18

**Review by salvogiangri — CHANGES_REQUESTED** [review]

File: target/a52q/patches/ascendia/module.prop (hunk 4 above)
Comment (RisenID) [reply target/a52q/patches/ascendia/module.prop]: Nah, im keeping u in there

File: target/a72q/patches/ascendia/customize.sh (hunk 2 above)
Comment (RisenID) [reply target/a72q/patches/ascendia/customize.sh]: done

File: target/a52q/patches/ascendia/customize.sh (hunk 1 above)
Comment (RisenID) [reply target/a52q/patches/ascendia/customize.sh]: done

File: target/a72q/patches/ascendia/customize.sh (hunk 5 above)
Comment (RisenID) [reply target/a72q/patches/ascendia/customize.sh]: Why? Im directly downloading the boot image and moving it, theres no extra stuff that needs to be moved.

edit: Ah i see what u mean nvm

Comment (RisenID) [reply target/a72q/patches/ascendia/customize.sh]: done

File: target/a52q/patches/ascendia/customize.sh (hunk 6 above)
Comment (RisenID) [reply target/a52q/patches/ascendia/customize.sh]: done

**Review by salvogiangri — APPROVED** [review]

**Pull request merged**